File: bin/BMB.r
##########################################################################
### Funcion Busqueda Multiarranque Basica
###     Lanza la Busqueda local un numero determinado de veces
###     para una solucion aleatoria generada, y se queda con la
###     mejor solucion de todas las encontradas
##########################################################################

BMB <- function(data){
  max.arranques <- BMB.num.sols.init
  
  # Aplicamos busqueda local un numero max.arranques veces
  masks <- lapply (1:max.arranques, function(x){ BL(data) })
  tasas <- sapply(masks, function(x){ tasa.clas(data, x) })
  
  masks[[which.max(tasas)]]
}

File: pscode/BMB.rb
def BMB(){
    # n el número de predictores de cada dataset
    mascaras = random_init(n)
    mascaras = [ BL(m): m in mascaras ]
    tasas = [ tasa.clas(m): m in mascaras ]

    return mascaras [arg max tasas]
}
